### DISCUSSION TITLE
MQTT Password Length (30 char) Limit / BUG?

### DISCUSSION THREAD

**Discussion by DECLLC** [meshtastic/Discussions#121] [Firmware]
Hi,

I'm trying to use MQTT with some other open source products over MQTT. The open source project pre-generates keys of 32 characters. Is there an easy fix for this on the Meshtastic side? It appears it's currently limited to 30 chars.

Thank you!

**Comment by garthvh** [comment]
The mqtt config is maxed out size wise so this is the largest the password can be

**Reply by DECLLC** [reply]
That's shame. Keycloak, used by OpenRemote, uses 32 bytes as its minimum length for shared keys. It doesn't seem that it would fit with their security focus to decrease this. Will see if I can modify it for testing in the db behind the scenes.

What does it mean when you say "mqtt config is maxed out"? 

I am guessing adding X.509 client cert support would require more changes and resources. 

Thank you

**Reply by garthvh** [reply]
Each config section has to stay under the max packet size for admin messages to be possible over the mesh, and mqtt is the largest config section.
